src/main/java/com/datadog/api/client/ApiClient.java

```diff
@@ -1562,6 +1591,31 @@ public <T> ApiResponse<T> invokeAPI(
     }
   }
 
+  private boolean shouldRetry(int retryCount, int statusCode, RetryConfig retryConfig) {
+    boolean statusToRetry = false;
+    if (statusCode == 429 || statusCode >= 500) {
+      statusToRetry = true;
+    }
+    return (retryConfig.maxRetries >= retryCount && statusToRetry && retryConfig.isEnableRetry());
```

Comment:
```suggestion
    return (retryConfig.maxRetries > retryCount && statusToRetry && retryConfig.isEnableRetry());
```
As you start with 0 this should be a strict comparison.

Reply:
I think it's still true?